app/src/main/java/com/example/campusguide/directions/indoor/IndoorPath.kt

```diff
@@ -0,0 +1,10 @@
+package com.example.campusguide.directions.indoor
+
+import com.example.campusguide.directions.Path
+import com.google.android.gms.maps.model.LatLng
+
+class IndoorPath(points: MutableList<LatLng>, val floor: Int) : Path(points) {
+    override fun shouldDisplay(currentFloor: Int): Boolean {
+        return currentFloor == floor
+    }
+}
```

Comment:
The problem i wanted to solve: only indoor paths need to decide whether they should display based on what the floor number is.
Rather than use an IF statement or something to check if it's an indoor path I took advantage of polymorphism.
So we need a whole class to make the code extensible (if we need more kinds of paths and reasons to not display) and to split responsibilities (not put an if chain in one class)